File: go-sdk/pkg/models/operations/getlegalholdpoliciesid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetLegalHoldPoliciesIDPathParams struct {
	LegalHoldPolicyID string `pathParam:"style=simple,explode=false,name=legal_hold_policy_id"`
}

type GetLegalHoldPoliciesIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetLegalHoldPoliciesIDPathParams
}

type GetLegalHoldPoliciesIDResponse struct {
	ClientError     *shared.ClientError
	ContentType     string
	LegalHoldPolicy *shared.LegalHoldPolicy
	StatusCode      int64
}

File: ts-sdk/src/sdk/models/operations/getlegalholdpoliciesid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetLegalHoldPoliciesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=legal_hold_policy_id" })
  legalHoldPolicyId: string;
}


export class GetLegalHoldPoliciesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetLegalHoldPoliciesIdPathParams;
}


export class GetLegalHoldPoliciesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  legalHoldPolicy?: shared.LegalHoldPolicy;

  @SpeakeasyMetadata()
  statusCode: number;
}
